File: primes.py
from sys import argv
print(end="2")
limit = int(argv[1])
half = limit // 2
sieve, total = bytearray({True}) * half, 1
for i in range(3, int(limit ** 0.5) + 1, 2):
	if sieve[i // 2]:
		for j in range(i ** 2 // 2, half, i):
			sieve[j] = False
for i in range(1, half):
	if sieve[i]:
		print(",", 2 * i + 1, end="")
		total += 1
print("\nTotal:", total)

File: primes.rb
# frozen_string_literal: true

print 2
limit = ARGV[0].to_i
half = limit / 2
sieve = "\x1" * half
total = 1
(3..Integer.sqrt(limit)).step(2).each do |i|
	next unless sieve[i / 2] == "\x1"

	(i**2 / 2...half).step(i).each do |j|
		sieve[j] = "\x0"
	end
end
(1...half).each do |i|
	if sieve[i] == "\x1"
		print ', ', 2 * i + 1
		total += 1
	end
end
puts "\nTotal: #{total}"
